Upstream: yes, was under review: https://codereview.qt-project.org/#/c/88074/
Reason: Fixes a crash triggered by Trojita

From f93fec1adcff7cb2afd1cbe4d3417983d9349257 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20Br=C3=BCning?= <michael.bruning@digia.com>
Date: Mon, 23 Jun 2014 12:17:14 +0200
Subject: [PATCH] Prevent crashes when the QWebView gets reparented.

Wrap the QWindow pointer in a QPointer so it gets set to NULL
when the QWindow gets deleted without notification.

Task-number: QTBUG-39524
Change-Id: I7c8fbe0f6f1921a7b48c58dc502d6a0a3409d084
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@digia.com>
---
 Source/WebKit/qt/WidgetApi/qwebpage_p.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/WebKit/qt/WidgetApi/qwebpage_p.h b/Source/WebKit/qt/WidgetApi/qwebpage_p.h
index 8ba6c4c..604c0a5 100644
--- a/Source/WebKit/qt/WidgetApi/qwebpage_p.h
+++ b/Source/WebKit/qt/WidgetApi/qwebpage_p.h
@@ -199,7 +199,7 @@ public:
 
     QAction *actions[QWebPage::WebActionCount];
 
-    QWindow* window;
+    QPointer <QWindow> window;
     QWidget* inspectorFrontend;
     QWebInspector* inspector;
     bool inspectorIsInternalOnly; // True if created through the Inspect context menu action
-- 
2.0.2

